calendar-bundle/src/Resources/contao/modules/ModuleEventReader.php

```diff
@@ -287,29 +290,14 @@ protected function compile()
 		$objTemplate->addBefore = false;
 
 		// Add an image
-		if ($objEvent->addImage && $objEvent->singleSRC)
+		if ($objEvent->addImage && null !== ($figureBuilder = $this->getFigureBuilderIfResourceExists($objEvent->singleSRC)))
 		{
-			$objModel = FilesModel::findByUuid($objEvent->singleSRC);
-
-			if ($objModel !== null && is_file(System::getContainer()->getParameter('kernel.project_dir') . '/' . $objModel->path))
-			{
-				// Do not override the field now that we have a model registry (see #6303)
-				$arrEvent = $objEvent->row();
-
-				// Override the default image size
-				if ($this->imgSize)
-				{
-					$size = StringUtil::deserialize($this->imgSize);
-
-					if ($size[0] > 0 || $size[1] > 0 || is_numeric($size[2]) || ($size[2][0] ?? null) === '_')
-					{
-						$arrEvent['size'] = $this->imgSize;
-					}
-				}
-
-				$arrEvent['singleSRC'] = $objModel->path;
-				$this->addImageToTemplate($objTemplate, $arrEvent, null, null, $objModel);
-			}
+			$figureBuilder
+				->setSize($this->imgSize ?: $objEvent->size ?: null)
```

Comment:
This would now use `$this->imgSize` even if it is set to an “empty” value like `a:3:{i:0;i:0;i:1;i:0;i:2;s:3:"box";}`. We should probably use an $imgSize variable like *ModuleEventlist.php* does in line 284.

Reply:
see 93f7e0376750f50ca7999490baea7f917de4cc87